### ISSUE TITLE
interface array loses it's contents

### ISSUE THREAD

**Issue by kahsa** [vlang/v#10260]
**V version:0.2.2 b29bc9c** 
**OS:ArchLinux** 

**What did you do?**

checked #7123 current status.

It used to work correctly, but now it loses the contents of the array

```v
struct Parent {
mut:
	children []Child
}

interface Child {
	add(int)
}

struct Taro {
mut:
	items []int
}

fn (mut t Taro) add(i int) {
	t.items << i
}

fn (mut p Parent) set() {
	mut taro := Taro{}
	taro.add(123)
	taro.add(456)
	taro.add(789)
	for i in taro.items {
		println('Taro added $i')
	}
	println('#1 ${taro.items}')
	p.children << taro
	println('#2 $p.children}')
}

// ver.1 not work -----

fn main() {
	mut parent := Parent{}
	parent.set()
	println('#3 ${parent.children}')
}
```

**What did you expect to see?**

ver.1 should work correctly as before.

**What did you see instead?** 

```
Taro added 123
Taro added 456
Taro added 789
#1 [123, 456, 789]
#2 [Child(Taro{
    items: [123, 456, 789]
})]}
#3 [Child(Taro{
    items: []
})]
```

**Comment by UweKrueger** [comment]
The problem is that `taro` in `Parent.set()` is allocated on the stack and using it as an interface takes the address without copying. Unfortunately, this address in no more valid after the `.set()` method has finished, since the stack is used otherwise. (It might have worked in some cases but that was more by accident.)

For now there are two solutions to make `taro` be allocated on the heap:

1. Use the `&` at creation of the object:
```v
    fn (mut p Parent) set() {
        mut taro := &Taro{} // `&` enforces heap allocation
        // ...
    }
```
2. Declare `struct Parent` to be *always* heap-allocated with the `[heap]` label:
```v
    [heap]
    struct Taro {
    mut:
    	items []int
    }
```

I'm currently working on making the compiler detect the situation and make an object automatically be allocated on the heap if used as interface, so you original example will work unchanged, soon.

**Comment by kahsa** [comment]
fixed by #10529
